### PR TITLE
MdeModulePkg: Consume SOC related ACPI table from ACPI Silicon HOB

### PR DESCRIPTION
**Pull request by GeorgeLiao012**
REF : https://bugzilla.tianocore.org/show_bug.cgi?id=4787

If ACPI Silicon Hob has been found from entry of AcpiTableDxe driver, that means SOC related ACPI tables been pass to the DXE phase by HOB. Each SOC related ACPI tables will be install.

# Description

<_Include a description of the change and why this change was made._>

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Will this cause a break in build or boot behavior?
  - Examples: Add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

<_Describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

File: MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c [hunk 1]
```diff
@@ -1995,6 +2098,23 @@ AcpiTableAcpiTableConstructor (
...
+    Status = InstallAcpiTableFromAcpiSiliconHob (AcpiTableInstance, GuidHob);
+    if (Status == EFI_SUCCESS) {
+      DEBUG ((DEBUG_INFO, "Installed ACPI Table from AcpiSiliconHob.\n"));
+      return EFI_SUCCESS;
+    } else {
+      DEBUG ((DEBUG_ERROR, "Fail to Installed ACPI Table from AcpiSiliconHob!!\n"));
+      ASSERT (Status != EFI_SUCCESS);
+    }
+  } else {
+    DEBUG ((DEBUG_ERROR, "Fail to locate AcpiSiliconHob!!\n"));
```
Comment (lgao4) [inline MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c:2115]: Here DEBUG level should be DEBUG_INFO intead of ERROR. Some platform boot doesn't depend on FSP.

File: MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTable.h [hunk 2]
```diff
@@ -126,6 +126,14 @@ typedef struct {
       EFI_ACPI_TABLE_SIGNATURE \
       )
 
+//
+// ACPI HOB produced by silicon initialization code will provide the RSDP structure.
+//
+typedef struct {
```
Comment (niruiyu) [inline MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTable.h:132]: 1. The HOB definition should be in a public header file.
2. Only "Rsdp" field is used by AcpiTableDxe code logic and the other two fields are not used at all.

File: MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c [hunk 3]
```diff
@@ -1947,6 +1947,109 @@ InstallAcpiTableFromHob (
...
+
+    if (SiCommonAcpiTable->Length <= sizeof (EFI_ACPI_DESCRIPTION_HEADER)) {
+      DEBUG ((DEBUG_ERROR, "RSDT length is incorrect\n"));
+      return EFI_ABORTED;
+    }
+    //
+    // Calcaue 32bit Acpi table number.
+    //
+    NumOfTblEntries = (SiCommonAcpiTable->Length - sizeof (EFI_ACPI_DESCRIPTION_HEADER)) / sizeof (UINT32);
+    AcpiTableInstance->NumberOfTableEntries1 = NumOfTblEntries;
```
Comment (niruiyu) [inline MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c:2014]: The FSP ACPI table is not relocated to an ACPI-reclaimed memory, instead, it is still in the FSP reserved memory range.
Is there an FSP spec change that reports the FSP ACPI table in correct (ACPI-reclaimed) memory type?

Comment (GeorgeLiao012) [reply MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c:2014]: As I know, FSP ACPI table PPI service is allocated EfiACPIReclaimMemory type memory for ACPI table under FSP phase, so ACPITableDxe can directly use ACPI table which from HOB.

**Pull request closed without merging**